src/Akeneo/Pim/Enrichment/Product/back/API/Command/UpsertProductCommand.php

```
@@ -26,28 +28,35 @@ public function __construct(
         private mixed $categoryUserIntent = null,
         private mixed $parentUserIntent = null,
         private mixed $groupsUserIntent = null,
-        private mixed $enabledUserIntent = null,
+        private ?SetEnabled $enabledUserIntent = null,
         private mixed $associationsUserIntent = null,
         private array $valueUserIntents = []
     ) {
         Assert::allImplementsInterface($this->valueUserIntents, ValueUserIntent::class);
     }
 
     /**
-     * @param ValueUserIntent[] $userIntents
+     * @param UserIntent[] $userIntents
      */
     public static function createFromCollection(int $userId, string $productIdentifier, array $userIntents): self
     {
         $valueUserIntents = [];
+        $enabledUserIntent = null;
+
         foreach ($userIntents as $userIntent) {
             if ($userIntent instanceof ValueUserIntent) {
                 $valueUserIntents[] = $userIntent;
             }
+
+            if ($userIntent instanceof SetEnabled) {
+                $enabledUserIntent = $userIntent;
```

Should we throw an exception if a SetEnabled was already defined?